### PR TITLE
support xz compression

### PR DESCRIPTION
**Pull request by jeffvance**
Fix issue #71 
Tested for local and remote imports. But need to add _local_ and _.xz_ tests. Will do that in a later commit here or in a separate pr.
So many new files due to `glide` and the xz repo vendored in.

### CODE REVIEW

File: pkg/image/decompress.go [hunk 1]
```diff
@@ -40,15 +41,23 @@ func UnpackData(filename string, src io.ReadCloser) (io.ReadCloser, error) {
 // return a ReadCloser.  It is up to the caller of DecompressData to close the returned stream.
 // If no compression is detected, it is considered a 'noop' and the original stream is returned.
 // Returns trimmed filename string and gzip Reader if gzip compression was used.
-func DecompressData(filename string, src io.ReadCloser) (string, io.ReadCloser, error) {
+func DecompressData(filename string, src io.ReadCloser) (fn string, rc io.ReadCloser, err error) {
```
Comment (copejon) [inline pkg/image/decompress.go]: Why use named returns here and not elsewhere?  I'm okay with going either way, so long as it's consistent.

File: pkg/image/decompress.go [hunk 2]
```diff
@@ -40,15 +41,23 @@ func UnpackData(filename string, src io.ReadCloser) (io.ReadCloser, error) {
...
 	glog.Infof("DecompressData: checking if %q is compressed\n", filename)
 	ext := filepath.Ext(TrimString(filename))
 	switch ext {
-	case ExtGz:
-		glog.Infof("DecompressData: detected %v compression format", ExtGz)
-		filename = strings.TrimSuffix(filename, ext) // trim ".gz"
-		src, err := gunzip(src)
-		return filename, src, err
+	case ExtGz, ExtXz:
```
Comment (copejon) [inline pkg/image/decompress.go]: The outer switch seems unnecessary. Why not 
```
switch ext {    
case ExtGZ:
        rc, err = gunzip(src)
case ExtXz:
	rc, err = xzDecompress(src)
default:
        rc = src
}
if err != nil {
    return "", nil, fmt.Errorf("DecompressData: %v\n", err)
}
fn = strings.TrimSuffix(filename, ext)
return fn, rc, nil
```

File: pkg/image/decompress.go [hunk 3]
```diff
@@ -71,6 +80,14 @@ func gunzip(r io.ReadCloser) (io.ReadCloser, error) {
 	return gzip.NewReader(r)
 }
 
+func xzDecompress(r io.ReadCloser) (io.ReadCloser, error) {
```
Comment (copejon) [inline pkg/image/decompress.go]: I like the name; should we change `gunzip` to match the pattern?  Like `gzDecompress`?  That helps denote that they both act similarly I think.

File: pkg/image/decompress.go (hunk 1 above)
Comment (jeffvance) [reply pkg/image/decompress.go]: It was to save a declaration of `var err error` within the scope of the outer switch. Go doesn't permit one returned value to be named while the other returned values are unnamed, so I named them all. 
I'll declare `err` explicitly.

File: pkg/image/decompress.go (hunk 2 above)
Comment (jeffvance) [reply pkg/image/decompress.go]: The outer switch is there to allow a single call to `glog.Infof()`.

**Comment by jeffvance** [comment]
I did some research on the _xi2/xz_ repo to get a better sense if it is reasonable for us to use.
Github `Insights` shows no activity in the past 3 months, no PRs, no issues, a single contributor (or company), no forks, and a flurry of commits back in Dec '17. So, overall, not great. However, the only commonly mentioned _xz_ repo is _ulikunitz/xz_, which is similar to _xi2/xz_ but with a longer period of inactivity. So there is some risk in using a non-golang package for xz decompression.

**Comment by copejon** [comment]
lgtm

**Pull request merged**